draft-ietf-dnsop-svcb-https.md

```diff
@@ -506,32 +503,37 @@ load-balancing.
 
 # Client behavior {#client-behavior}
 
-An SVCB-aware client resolves an origin HOST by attempting to determine
+An SVCB-aware client resolves a service with hostname $HOST by attempting
+to determine
 the preferred SvcFieldValue and IP addresses for its service, using the
 following procedure:
 
-1. Issue parallel AAAA/A and SVCB queries for the name HOST.
+1. Let $SVCB_QNAME be $HOST, plus appropriate prefixes for the scheme
+   (see {{svcb-names}}).
+
+2. Issue parallel AAAA/A queries for $HOST and a SVCB query for $SVCB_QNAME.
    The answers for these may or may not include CNAME pointers
    before reaching one or more of these records.
 
-2. If a SVCB record of AliasForm SvcRecordType is returned for HOST,
-   clients MUST loop back to step 1 replacing HOST with SvcDomainName,
+3. If a SVCB record of AliasForm SvcRecordType is returned for $SVCB_QNAME,
+   clients MUST set $HOST and $SVCB_QNAME to SvcDomainName (without
+   additional prefixes) and loop back to step 2,
    subject to chain length limits and loop detection heuristics (see
    {{client-failures}}).
 
-3. If one or more SVCB records of ServiceForm SvcRecordType are returned
-   for HOST, clients should select the highest-priority option with
+4. If one or more SVCB records of ServiceForm SvcRecordType are returned
+   for $SVCB_QNAME, clients should select the highest-priority option with
    acceptable parameters, and resolve AAAA and/or A records for its
    SvcDomainName if they are not already available.  These are the
    preferred SvcFieldValue and IP addresses.  If the connection fails, the
    client MAY try to connect using values from a lower-priority record.
-   If none of the options succeed, the client SHOULD connect to the origin
-   server directly.
+   If none of the options succeed, the client SHOULD connect to the authority
+   address directly.
```

Comment:
Should it connect to the authority address or $HOST_QNAME?  The latter implies that the AliasForm SVCB record should be ignored if ServiceForm SVCB records were present and didn't work.

Reply:
(This is an interesting question which may want to be its own issue for discussion.)